### PR TITLE
sdk(react-native-sdk): created script that compares dep with peerDep versions

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react-native-sdk/update_dependencies.js [hunk 1]
```diff
@@ -21,6 +24,13 @@ function updateDependencies() {
             packageJSON.dependencies[key] = RNSDKpackageJSON.peerDependencies[key];
             updated = true;
         }
+
+        if (semver.satisfies(RNSDKpackageJSON.peerDependencies[key], `> ${packageJSON.dependencies[key]}`)) {
```
Comment (saghul) [inline react-native-sdk/update_dependencies.js:28]: Do we need to put the `>` in there? What if they are equal?

Comment (saghul) [inline react-native-sdk/update_dependencies.js:28]: If it doesn't saisfy the requirement we need do something about it: log something, tell the user...

Comment (Calinteodor) [reply react-native-sdk/update_dependencies.js:28]: That is what we need to check. If that condition is satisfied.

File: react-native-sdk/update_dependencies.js [hunk 2]
```diff
@@ -21,6 +24,19 @@ function updateDependencies() {
             packageJSON.dependencies[key] = RNSDKpackageJSON.peerDependencies[key];
             updated = true;
         }
+
+        if (semver.satisfies(RNSDKpackageJSON.peerDependencies[key], `=${packageJSON.dependencies[key]}`)) {
+            return;
```
Comment (saghul) [inline react-native-sdk/update_dependencies.js:29]: You want to keep checking other packages no? So `continue` then.

File: react-native-sdk/update_dependencies.js [hunk 3]
```diff
@@ -21,6 +24,19 @@ function updateDependencies() {
...
+        }
+
+        if (semver.satisfies(RNSDKpackageJSON.peerDependencies[key], `>${packageJSON.dependencies[key]}`)) {
```
Comment (saghul) [inline react-native-sdk/update_dependencies.js:32]: Why do we need to update it if does satisfy the requirement? Isn't thjis backwards?

File: react-native-sdk/update_dependencies.js [hunk 4]
```diff
@@ -21,6 +24,19 @@ function updateDependencies() {
...
+            packageJSON.dependencies[key] = RNSDKpackageJSON.peerDependencies[key];
+            updated = true;
+
+            console.log(`${key} is now set to ${RNSDKpackageJSON.peerDependencies[key]}`);
+        } else {
+            updated = false;
```
Comment (saghul) [inline react-native-sdk/update_dependencies.js:38]: Don't reset the flag, other packages might have been updated.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react-native-sdk/update_dependencies.js (hunk 3 above)
Comment (Calinteodor) [reply react-native-sdk/update_dependencies.js:32]: No, it's not. If our peer dependencies have a bigger version than the dependencies that are in the app that is using our rnsdk, then we need to update.

File: react-native-sdk/update_dependencies.js (hunk 2 above)
Comment (Calinteodor) [reply react-native-sdk/update_dependencies.js:29]: Why continue? We are comparing all of our peer dependencies with the dependencies that have the same property. If they all have the same version, that's what we expect and we stop.

File: react-native-sdk/update_dependencies.js (hunk 3 above)
Comment (saghul) [reply react-native-sdk/update_dependencies.js:32]: Ah right!

**Review by saghul — DISMISSED** [review]

**Review by saghul — DISMISSED** [review]

**Review by saghul — APPROVED** [review]

**Pull request merged**